### PR TITLE
fix: allow flutter run to take cli arguments

### PR DESCRIPTION
**Pull request by joshpetit**
#118 

Fix flutter run not using arguments.

This wont work for emulators, I think I can get it to work by just extending the args again and then using the run method instead of emulator run, but im not sure if that might break something else.

### CODE REVIEW

File: lua/flutter-tools/devices.lua [hunk 1]
```diff
@@ -100,7 +100,12 @@ function M.select_device()
     if device.type == EMULATOR then
       M.launch_emulator(device)
     else
-      require("flutter-tools.commands").run({ device = device })
+      if args then
+        vim.list_extend(args, { "-d", device.id })
+        require("flutter-tools.commands").run({ full_args = args })
```
Comment (akinsho) [inline lua/flutter-tools/devices.lua:105]: same here please 🙏🏿

**Review by akinsho — CHANGES_REQUESTED** [review]
Thanks for this PR 👍🏿 I've made some minor comments to tweak variable names but it looks good in general

File: lua/flutter-tools/commands.lua [hunk 2]
```diff
@@ -126,22 +130,25 @@ function M.run(opts)
   opts = opts or {}
   local device = opts.device
   local cmd_args = opts.args
+  local full_args = opts.full_args
```
[the remark this answers is not in the document]
Comment (joshpetit) [reply lua/flutter-tools/commands.lua]: Yea that's a good choice 👍👍

**Comment by joshpetit** [comment]
@akinsho just fixed the pipeline, I was using a different formatter before.

**Review by akinsho — APPROVED** [review]
Thanks for the tweaks @joshpetit, this LGTM haven't tested it out myself but should be fine 👍🏿 🤞🏿

**Pull request merged**